rules/integrations/aws/impact_s3_bucket_object_uploaded_with_ransom_extension.toml

```diff
@@ -77,44 +103,38 @@ tags = [
     "Resources: Investigation Guide",
 ]
 timestamp_override = "event.ingested"
-type = "esql"
+type = "eql"
 
 query = '''
-from logs-aws.cloudtrail-*
-
-// any successful uploads via S3 API requests
-| where
-  event.dataset == "aws.cloudtrail"
-  and event.provider == "s3.amazonaws.com"
-  and event.action == "PutObject"
-  and event.outcome == "success"
-
-// extract object key from API request parameters
-| dissect aws.cloudtrail.request_parameters "%{?ignore_values}key=%{Esql.aws_cloudtrail_request_parameters_object_key}}"
-
-// regex match against common ransomware naming patterns
-| where
-  Esql.aws_cloudtrail_request_parameters_object_key rlike "(.*)(ransom|lock|crypt|enc|readme|how_to_decrypt|decrypt_instructions|recovery|datarescue)(.*)"
-  and not Esql.aws_cloudtrail_request_parameters_object_key rlike "(.*)(AWSLogs|CloudTrail|access-logs)(.*)"
-
-// keep relevant ECS and derived fields
-| keep
-  tls.client.server_name,
-  aws.cloudtrail.user_identity.arn,
-  Esql.aws_cloudtrail_request_parameters_object_key
-
-// aggregate by server name, actor, and object key
-| stats
-    Esql.event_count = count(*)
-  by
-    tls.client.server_name,
-    aws.cloudtrail.user_identity.arn,
-    Esql.aws_cloudtrail_request_parameters_object_key
-
-// filter for rare single uploads (likely test/detonation)
-| where Esql.event_count == 1
+file where
+  event.dataset == "aws.cloudtrail" and
+  event.provider == "s3.amazonaws.com" and
+  event.action == "PutObject" and
+  event.outcome == "success" and
+  /* Apply regex to match patterns only after the bucket name */
+  target.entity.id regex "arn:aws:s3:::[^/]+/.*?(ransom|lock|crypt|enc|readme|how_to_decrypt|decrypt_instructions|recovery|datarescue).*" and
+  not target.entity.id regex ".*(AWSLogs|CloudTrail|access-logs).*"
 '''
 
+[rule.investigation_fields]
+field_names = [
+    "@timestamp",
+    "user.name",
+    "user_agent.original",
+    "source.ip",
+    "aws.cloudtrail.user_identity.arn",
+    "aws.cloudtrail.user_identity.type",
+    "aws.cloudtrail.user_identity.access_key_id",
+    "aws.cloudtrail.resources.arn",
+    "aws.cloudtrail.resources.type",
+    "tls.client.server_name",
+    "event.action",
+    "event.outcome",
+    "cloud.account.id",
+    "cloud.region",
+    "aws.cloudtrail.request_parameters",
+    "aws.cloudtrail.response_elements"
```

Comment:
Do we want to add target.entity.id here?

Reply:
`aws.cloudtrail.resources.arn` captures the same information, and actually I am going to change the query to use this field so the rule will be compatible with filebeat. Thank you for this comment